### PR TITLE
feat(checkbox): add base structure for use outside ion-item

### PR DESCRIPTION
**Pull request by amandaejohnston**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
  - Some docs updates need to be made in the `ionic-docs` repo, in a separate PR. See the [contributing guide](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#modifying-documentation) for details.
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Checkbox needs to be converted to use the new standalone form component syntax.

Issue URL: N/A

## What is the new behavior?

- Added stub functionality for rendering under the new syntax. (Styles and additional props will be sorted out in separate PRs.)
- Added `legacy` prop.
- Copied old tests to `legacy` folder and updated new versions to use the new syntax.
- Removed "clickable item" from top of `basic` test and moved "unchecked by default" item to the top in its place. (The click handler on that clickable item didn't actually exist, so it was throwing console errors.)
- Removed nested checkmarks from `indeterminate` test as they were testing custom functionality and thus not needed, plus they were causing console errors whenever the checkboxes were clicked.
- Adjusted legacy `indeterminate` test to avoid extra white space at the bottom of the screenshots.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: core/src/components/checkbox/checkbox.tsx [hunk 1]
```diff
@@ -144,6 +170,35 @@ export class Checkbox implements ComponentInterface {
...
+        `Using ion-checkbox with an ion-label has been deprecated. To migrate, remove the ion-label and pass your label directly into ion-checkbox instead.
+Example: <ion-checkbox>Email:</ion-checkbox>
+For checkboxes that do not have a visible label, developers should use "aria-label" so screen readers can announce the purpose of the checkbox.`,
+        this.el
+      );
+
+      if (this.legacy) {
+        printIonWarning(
+          `ion-checkbox is being used with the "legacy" property enabled which will forcibly enable the legacy form markup. This property will be removed in an upcoming major release of Ionic where this form control will use the modern form markup.
+Developers can dismiss this warning by removing their usage of the "legacy" property and using the new input syntax.`,
```
Comment (sean-perkins) [inline core/src/components/checkbox/checkbox.tsx:194]: Should this read "new checkbox syntax"?

File: core/src/utils/forms/form-controller.ts [hunk 2]
```diff
@@ -4,7 +4,7 @@
  * @internal
  * @prop el: The Ionic form component to reference
  */
-type AllowedFormElements = HTMLIonInputElement | HTMLIonToggleElement;
+type AllowedFormElements = HTMLIonCheckboxElement | HTMLIonInputElement | HTMLIonToggleElement;
```
Comment (sean-perkins) [inline core/src/utils/forms/form-controller.ts:7]: You may want to take what I have done here: https://github.com/ionic-team/ionic-framework/blob/FW-2764-placement/core/src/utils/forms/form-controller.ts#L1 so we don't need to keep adding additional elements to the type here.

**Review by sean-perkins — COMMENTED** [review]
The image test being flaky should be resolved once: https://github.com/ionic-team/ionic-framework/pull/26440 is merged. 

The picker-internal we might need to create a tech debt ticket to track. If you re-run the update reference screenshot job, I've had mixed success of it capturing correctly (though we should still try to fix it).

The textarea autogrow is a strange one, appears to just be a text rendering inconsistency.

File: core/src/components/checkbox/checkbox.tsx (hunk 1 above)
Comment (amandaejohnston) [reply core/src/components/checkbox/checkbox.tsx:194]: Good catch 👀

**Review by sean-perkins — APPROVED** [review]

**Review by liamdebeasi — APPROVED** [review]

**Pull request merged**